lib/index.js

```diff
@@ -432,6 +432,71 @@ api.setRoles = function(actor, identity, callback) {
 };
 
 /**
+ * Updates an Identity's ResourceRole.
+ *
+ * @param actor the Identity performing the action.
+ * @param options the options to use:
+ *   // FIXME: update doc
+ * @param callback(err) called once the operation completes.
+ */
+api.updateRole = function(actor, options, callback) {
+  if(!options || typeof options !== 'object') {
+    throw new TypeError('options must be specified.');
+  }
+  if(typeof options.role !== 'string') {
+    throw new TypeError('options.role must be a string.');
+  }
+  if(typeof options.identityId !== 'string') {
+    throw new TypeError('options.identityId must be a string.');
+  }
+  if(typeof options.operation !== 'string') {
+    throw new TypeError('options.operation must be a string.');
+  }
+  if(!['add', 'delete'].includes(options.operation)) {
```

Comment:
Maybe we should have `CAPABILITY_ADD_ROLE` and `CAPABILITY_ADD_RESOURCE` to go along with `CAPABILITY_REMOVE_ROLE` and `CAPABILITY_REMOVE_RESOURCE` instead of "assign". And then the two new roles we create each have ADD+REMOVE permissions, respectively.